examples/with-videojs/components/Player.js

```diff
@@ -1,34 +1,23 @@
-import { Component } from 'react'
+import { useRef, useEffect } from 'react'
 import videojs from 'video.js'
 import 'videojs-youtube'
 
-class Player extends Component {
-  componentDidMount() {
-    // instantiate Video.js
-    this.player = videojs(this.videoNode, this.props, function onPlayerReady() {
-      console.log('onPlayerReady', this)
-    })
-  }
+const Player = (props) => {
+  const playerRef = useRef()
 
-  // destroy player on unmount
-  componentWillUnmount() {
-    if (this.player) {
-      this.player.dispose()
+  useEffect(() => {
+    const player = videojs(playerRef.current, props)
+
+    return () => {
+      player.dispose()
     }
-  }
+  }, [player])
```

Comment:
This should be `playerRef`, right?
```suggestion
  }, [playerRef])
```

Reply:
correct

Reply:
Also, this needs to use a callback ref AFAIK, not `useRef` (i.e. `useCallback`).